File: lab/python/somas/soma_digitos_recursivo.py
def soma_digitos_recursivo(n):
    n = abs(n)  # Considera o valor absoluto para lidar com números negativos
    if n == 0:
        return 0
    else:
        return n % 10 + soma_digitos_recursivo(n // 10)


print(soma_digitos_recursivo(1234))  # Saída: 10
print(soma_digitos_recursivo(-567))   # Saída: 18

File: lab/c/somas/soma_digitos_recursivo.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>


int soma_digitos(int n) {
    n = abs(n); // Considera o valor absoluto para lidar com números negativos
    if (n < 10) {
        return n;
    } else {
        return (n % 10) + soma_digitos(n / 10);
    }
}

int main() {
    int numero;
    printf("Digite um número inteiro: ");
    scanf("%d", &numero);
    
    int resultado = soma_digitos(numero);
    printf("A soma dos dígitos de %d é: %d\n", numero, resultado);
    
    return 0;
}
